### PR TITLE
 [ACA-4361] permission layout modified

### PR DESCRIPTION
**Pull request by dhrn**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://alfresco.atlassian.net/browse/ACA-4361

**What is the new behaviour?**

fixed

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — APPROVED** [review]

File: lib/content-services/src/lib/permission-manager/components/user-name-column/user-name-column.component.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        if (this.context) {
+            this.updateContextValue();
+        }
+
+        if (this.node) {
+            this.updateNodeValue();
+        }
+    }
+
+    protected updateValue(person: Person, group: Group) {
```
Comment (davidcanonieto) [inline lib/content-services/src/lib/permission-manager/components/user-name-column/user-name-column.component.ts:57]: could we simplify this logic, after all it's one or the other, always one of them is going to be null.

File: lib/content-services/src/lib/permission-manager/components/user-name-column/user-name-column.component.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if (authorityId) {
+            this.updateValue(null, { displayName: authorityId } as any);
+        } else {
+            this.updateValue(person, group);
+        }
+    }
+
+    private updateNodeValue() {
+        const { entry } = this.node;
+        const person = this.nodePermissionService.transformNodeToPerson(entry);
```
Comment (davidcanonieto) [inline lib/content-services/src/lib/permission-manager/components/user-name-column/user-name-column.component.ts:79]: And same here, could we have a smart method that returns the corresponding Person/Group based on the node passed?

**Review by davidcanonieto — CHANGES_REQUESTED** [review]

File: lib/content-services/src/lib/permission-manager/components/add-permission/add-permission-panel.component.html [hunk 3]
```diff
@@ -33,15 +33,12 @@
             class="adf-permission-result-list"
             [class.adf-permission-result-list-search]="searchedWord.length === 0">
     <ng-template let-data>
-        <mat-selection-list class="adf-permission-result-list-elements">
+        <mat-selection-list class="adf-permission-result-list-elements" (keydown.control.a)="selectAll( data?.list?.entries)">
```
[the remark this answers is not in the document]
Comment (dhrn) [reply lib/content-services/src/lib/permission-manager/components/add-permission/add-permission-panel.component.html:36]: yes it add all the result to selected list. it is not paginated (array of node entries from search result ).

File: lib/content-services/src/lib/permission-manager/components/add-permission/add-permission.component.ts [hunk 4]
```diff
@@ -74,4 +81,13 @@ export class AddPermissionComponent implements OnInit {
         }
     }
 
+    private transformNodeToPermissionElement(nodes: NodeEntry[], nodeRole: any): PermissionElement[] {
```
[the remark this answers is not in the document]
Comment (dhrn) [reply lib/content-services/src/lib/permission-manager/components/add-permission/add-permission.component.ts:84]: it is string.. wrong type i will update it

File: e2e/content-services/pages/permissions.page.ts [hunk 5]
```diff
@@ -15,29 +15,97 @@
...
+} from '@alfresco/adf-testing';
 import { browser, by, element } from 'protractor';
 
 export class PermissionsPage {
 
     dataTableComponentPage = new DataTableComponentPage();
     addPermissionsDialog = new AddPermissionsDialogPage();
 
+    rootElement = TestElement.byId('adf-permission-manager-card');
+    emptyElement = new TestElement(this.rootElement.elementFinder.element(by.css('.adf-no-permission__template')));
```
Comment (adinapitul) [inline e2e/content-services/pages/permissions.page.ts:34]: this looks weird. Why is this element so complicated?

is something like this not working?
```
emptyElement = TestElement.byCss('.adf-no-permission__template');
```
or
```
rootSelector = 'adf-permission-manager-card';
emptyElement = TestElement.byCss(`${this.rootSelector} .adf-no-permission__template'`);
```

File: e2e/content-services/pages/permissions.page.ts [hunk 6]
```diff
@@ -15,29 +15,97 @@
...
+        const row = this.dataTableComponentPage.getRow('Users and Groups', name);
+        await row.click();
+        await BrowserActions.click(row.element(by.css('[id="adf-select-role-permission"] .mat-select-trigger')));
+    }
+
+    async clickDeletePermissionButton(username: string): Promise<void> {
+        const userOrGroupName = TestElement.byCss(`[data-automation-id="adf-delete-permission-button-${username}"]`);
+        await userOrGroupName.waitPresent();
+        await userOrGroupName.click();
+
```
Comment (adinapitul) [inline e2e/content-services/pages/permissions.page.ts:72]: remove empty line

File: e2e/content-services/pages/permissions.page.ts [hunk 7]
```diff
@@ -15,29 +15,97 @@
...
+        const userOrGroupName = TestElement.byCss('div[data-automation-id="' + username + '"]');
+        await userOrGroupName.waitNotPresent();
+    }
+
+    async noPermissionContent(): Promise<string> {
+        const noPermission = TestElement.byCss('.adf-no-permission__template--text');
+        return noPermission.getText();
+    }
+
+    async waitVisible(): Promise<void> {
```
Comment (adinapitul) [inline e2e/content-services/pages/permissions.page.ts:85]: "visible" and "present" are not the same. Method name is misleading:waitVisible, when in fact inside you're waiting for presence.

File: e2e/content-services/pages/permissions.page.ts [hunk 8]
```diff
@@ -15,29 +15,97 @@
...
+        await this.rootElement.waitPresent();
+    }
+
+    async isErrored(): Promise<boolean> {
```
Comment (adinapitul) [inline e2e/content-services/pages/permissions.page.ts:89]: looking at the element name and selector, "emptyElement", ".adf-no-permission__template", I don't understand the name of this method "isErrored", or the name of the following method: "waitForError"
please find better / more relevant naming.
If ".adf-no-permission__template" really is an error, then the element should not be called "emptyElement"

File: e2e/content-services/pages/permissions.page.ts [hunk 9]
```diff
@@ -15,29 +15,97 @@
...
+    }
+
+    async waitForError(): Promise<void> {
+        await this.emptyElement.waitPresent();
+    }
+
+    async waitTillContentLoads(): Promise<void> {
         await browser.sleep(500);
-        await this.dataTableComponentPage.checkRowIsNotSelected('Authority ID', name);
+        const loader = new TestElement(this.rootElement.elementFinder.element(by.css('adf-permission-loader')));
```
Comment (adinapitul) [inline e2e/content-services/pages/permissions.page.ts:99]: please simplify the loader element definition. I'm sure there's a way without all this rootElement.elementFinder.element by.css .....

File: e2e/content-services/pages/permissions.page.ts [hunk 10]
```diff
@@ -15,29 +15,89 @@
...
+    async noPermissionContent(): Promise<string> {
+        const noPermission = TestElement.byCss('.adf-no-permission__template--text');
+        return noPermission.getText();
+    }
+
+    async waitVisible(): Promise<void> {
+        await TestElement.byId(this.rootElement).waitVisible();
+    }
+
+    async waitTillContentLoads(): Promise<void> {
```
Comment (adinapitul) [inline e2e/content-services/pages/permissions.page.ts:89]: what's the purpose of isNotDisplayed? you're only using it for the log message? 
on the catch block, if isNotDisplayed is false, does this mean **IT IS** displayed? (I don't know who's "it" in this :) )

can you please explain what this method does?

File: e2e/content-services/components/permissions-component.e2e.ts [hunk 11]
```diff
@@ -207,8 +215,9 @@ describe('Permissions Component', () => {
         });
 
         it('[C274691] Should be able to add a new User with permission to the file and also change locally set permissions', async () => {
-            await expect(await permissionsPage.addPermissionsDialog.getRoleCellValue(filePermissionUser.username)).toEqual('Contributor');
-            await permissionsPage.addPermissionsDialog.clickRoleDropdownByUserOrGroupName(filePermissionUser.username);
+            await expect(await permissionsPage.getRoleCellValue(filePermissionUser.username)).toEqual('Contributor');
+            await permissionsPage.clickRoleDropdownByUserOrGroupName(filePermissionUser.username);
+            await browser.sleep(500);
```
[the remark this answers is not in the document]
Comment (adinapitul) [reply e2e/content-services/components/permissions-component.e2e.ts:220]: sleep is still here. Why is it needed? Can you instead wait for something specific? for an element to be visible or not.... or something...

**Pull request merged**